sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/pubsub/PubsubSchemaTransformWriteConfiguration.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.io.gcp.pubsub;
+
+import com.google.auto.value.AutoValue;
+import javax.annotation.Nullable;
+import org.apache.beam.sdk.annotations.Experimental;
+import org.apache.beam.sdk.schemas.AutoValueSchema;
+import org.apache.beam.sdk.schemas.Schema;
+import org.apache.beam.sdk.schemas.annotations.DefaultSchema;
+
+/**
+ * Configuration for writing to Pub/Sub.
+ *
+ * <p><b>Internal only:</b> This class is actively being worked on, and it will likely change. We
+ * provide no backwards compatibility guarantees, and it should not be implemented outside the Beam
+ * repository.
+ */
+@Experimental
+@DefaultSchema(AutoValueSchema.class)
+@AutoValue
+public abstract class PubsubSchemaTransformWriteConfiguration {
+
+  /** The expected schema of the Pub/Sub message. */
+  public abstract Schema getDataSchema();
+
+  /**
+   * The topic to which to write Pub/Sub messages.
+   *
+   * <p>See {@link PubsubIO.PubsubTopic#fromPath(String)} for more details on the format of the
+   * topic string.
+   */
+  public abstract String getTopic();
+
+  /**
+   * The expected format of the Pub/Sub message.
+   *
+   * <p>Used to retrieve the {@link org.apache.beam.sdk.schemas.io.payloads.PayloadSerializer} from
+   * {@link org.apache.beam.sdk.schemas.io.payloads.PayloadSerializers}.
+   */
+  @Nullable
+  public abstract String getFormat();
+
+  /** Used by the ProtoPayloadSerializerProvider when serializing to a Pub/Sub message. */
+  @Nullable
+  public abstract String getProtoClass();
+
+  /** Used by the ThriftPayloadSerializerProvider when serializing to Pub/Sub message. */
+  @Nullable
+  public abstract String getThriftClass();
+
+  /** Used by the ThriftPayloadSerializerProvider when serializing to Pub/Sub message. */
+  @Nullable
+  public abstract String getThriftProtocolFactoryClass();
```

Comment:
I'm wondering if we should just avoid using these. Since these transforms will be mostly configured by commands/UIs, I don't expect users to fiddle with serialization format too much. I would think Format would only be JSON or Avro. Thoughts?